src/components/editor/property/InputLookup.jsx

```diff
@@ -142,28 +148,98 @@ const InputLookup = (props) => {
 
   let error
   let groupClasses = 'form-group'
+  const classes = ['modal', 'fade']
+  let display = 'none'
+  const modalId = `InputLookupModal-${props.property.key}`
 
   if (displayValidations && !_.isEmpty(errors)) {
     groupClasses += ' has-error'
     error = errors.join(',')
   }
+
+  if (props.show) {
+    classes.push('show')
+    display = 'block'
+  }
+
+  const handleClick = (event) => {
+    event.preventDefault()
+    dispatch(showModal(modalId))
+  }
+
+  const close = (event) => {
+    props.hideModal()
+    event.preventDefault()
+  }
+
+  // TODO: New styling to fit description in #2478
+  const lookupSelection = props.lookupValues.map((lookupValue) => {
+    let lookupLink = ''
+    if (lookupValue.uri) { lookupLink = `<a href=${lookupValue.uri}><span aria-hidden="true"><FontAwesomeIcon className="globe-icon" icon={faGlobe} /></a>` }
+
+    return (
+      <div key={lookupValue.key} class="lookup-value">
+        <span key={lookupValue.key}>{lookupValue.label || lookupValue.literal}</span>
+        <button
+          onClick={() => props.removeValue(lookupValue.key)}
+          aria-label={`Remove ${lookupValue.label}`}
+          data-testid={`Remove ${lookupValue.label}`}
+          className="close rbt-close rbt-token-remove-button">
+        <span aria-hidden="true"><FontAwesomeIcon className="trash" icon={faTrashAlt} /></span>
+        </button>        
+        <a href={lookupValue.uri}>
+          <span aria-hidden="true"><FontAwesomeIcon className="globe-icon" icon={faGlobe} /></span>
+        </a>
+      </div>
+    )
+  })
+
+  const modal = (
```

Comment:
See previous comment about possible extracting this modal into a separate component.